submissions/Kostiantyn_Rakhmanov/frogger/app.js

```diff
@@ -0,0 +1,116 @@
+// Enemies our player must avoid
+const X_ENEMY = {
+    min: -100,
+    max: 510,
+};
+
+const ENEMY_SPEED = {
+    min: 50,
+    max: 100,
+};
+const ENEMY_LOCATIONS_Y = [60, 145, 225];
+
+const SECTION = {
+    width: 100,
+    height: 80,
+};
+
+const CANVAS = {
+    start: 0,
+    rows: 5,
+    columns: 4,
+};
+
+const CANVAS_WIDTH = SECTION.width * CANVAS.columns;
+const CANVAS_HEIGHT = SECTION.height * CANVAS.rows;
+
+const PLAYER_COORDS = {
+    x: 200,
+    y: 375,
+};
+const PLAYER_SHIFT = 60;
+const NEW_PLAYER_DELAY = 777;
+
+
+let Enemy = function(x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+
+    this.x += this.speed * dt;
+
+    if (this.x > X_ENEMY.max) {
+        this.x = X_ENEMY.min;
+        this.speed = ENEMY_SPEED.min + Math.floor(Math.random() * ENEMY_SPEED.max);
+    }
+    if (player.x < this.x + SECTION.height &&
+        player.x + SECTION.height > this.x &&
+        player.y < this.y + PLAYER_SHIFT &&
+        PLAYER_SHIFT + player.y > this.y) {
+            player.x = PLAYER_COORDS.x;
+            player.y = PLAYER_COORDS.y;
```

Comment:
`player` is a global variable. Requirements address this specific issue and also provide a hint on resolution. You may pass whatever you want to class constructor and store it as a class property.